beacon-chain/attestation/service_test.go

```diff
@@ -4,130 +4,41 @@ import (
 	"context"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
 	"github.com/prysmaticlabs/prysm/beacon-chain/types"
-	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
-	"github.com/prysmaticlabs/prysm/shared/database"
 	"github.com/prysmaticlabs/prysm/shared/testutil"
 	logTest "github.com/sirupsen/logrus/hooks/test"
 )
 
-func TestStartStop(t *testing.T) {
-	hook := logTest.NewGlobal()
+func setupService(t *testing.T) *Service {
 	ctx := context.Background()
 
-	config := &database.DBConfig{DataDir: "", Name: "", InMemory: true}
-	db, err := database.NewDB(config)
+	config := db.Config{Path: "", Name: "", InMemory: true}
+	db, err := db.NewDB(config)
 	if err != nil {
 		t.Fatalf("could not setup beaconDB: %v", err)
 	}
 
-	AttestHandler, err := NewHandler(db.DB())
-	if err != nil {
-		t.Fatalf("could not get attestation handler: %v", err)
-	}
-
 	cfg := &Config{
-		Handler: AttestHandler,
-	}
-
-	attestService := NewAttestService(ctx, cfg)
-
-	attestService.IncomingAttestationFeed()
-	attestService.ContainsAttestation([]byte{}, [32]byte{})
-
-	// Test the start function.
-	attestService.Start()
-
-	// Test the stop function.
-	if err := attestService.Stop(); err != nil {
-		t.Fatalf("unable to attest chain service: %v", err)
-	}
-
-	// The context should have been canceled.
-	if attestService.ctx.Err() == nil {
-		t.Error("context was not canceled")
+		BeaconDB: db,
 	}
 
-	testutil.AssertLogsContain(t, hook, "Starting service")
-	testutil.AssertLogsContain(t, hook, "Stopping service")
+	return NewAttestationService(ctx, cfg)
 }
 
 func TestIncomingAttestations(t *testing.T) {
 	hook := logTest.NewGlobal()
-	config := &database.DBConfig{DataDir: "", Name: "", InMemory: true}
-	db, err := database.NewDB(config)
-	if err != nil {
-		t.Fatalf("could not setup beaconDB: %v", err)
-	}
-
-	attestationHandler, err := NewHandler(db.DB())
-	if err != nil {
-		t.Fatalf("could not register blockchain service: %v", err)
-	}
-
-	cfg := &Config{
-		ReceiveAttestationBuf: 0,
-		Handler:               attestationHandler,
-	}
-
-	attestationService := NewAttestService(context.Background(), cfg)
+	service := setupService(t)
 
 	exitRoutine := make(chan bool)
 	go func() {
-		attestationService.aggregateAttestations()
+		service.aggregateAttestations()
 		<-exitRoutine
 	}()
 
-	attestationService.incomingChan <- types.NewAttestation(nil)
-	attestationService.cancel()
+	service.incomingChan <- types.NewAttestation(nil)
+	service.cancel()
 	exitRoutine <- true
 
 	testutil.AssertLogsContain(t, hook, "Forwarding aggregated attestation")
 }
-
-func TestContainsAttestations(t *testing.T) {
```

Comment:
You'll notice that I removed some tests. I get the feeling that we have too many complicated tests in the `service` layer, and I'd like to try and limit them. That having been said, I'm going to take another look at the tests that I removed tomorrow morning.